packages/web-components/src/components/footer/language-selector.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, query } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import BXComboBox from 'carbon-web-components/es/components/combo-box/combo-box.js';
+import BXComboBoxItem from 'carbon-web-components/es/components/combo-box/combo-box-item.js';
+import { FOOTER_SIZE } from './footer';
+import styles from './footer.scss';
+import { findIndex, forEach } from '../../globals/internal/collection-helpers';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Language selector
+ *
+ * @element dds-language-selector
+ */
+@customElement(`${ddsPrefix}-language-selector`)
+class DDSLanguageSelector extends BXComboBox {
+  @property()
+  lastValidLang = 'English';
+
+  /**
+   * Size property to apply different styles.
+   */
+  @property()
+  footerSize = FOOTER_SIZE.REGULAR;
+
+  /**
+   * The `<input>` for filtering.
+   */
+
+  @query('input')
+  private filterInputNode!: HTMLInputElement;
+
+  /**
+   * The shadow slot this language-selector should be in.
+   */
+
+  @property({ reflect: true })
+  slot = 'language-selector';
+
+  /**
+   * Reverts input value to last chosen valid language.
+   *
+   * @param e MouseEvent
+   */
+
+  protected _handleOutsideClick = e => {
```

Comment:
See other files using `@HostListener` to see how to use `@HostListener`. It involves usage of `HostListenerMixin` as well as importing the decorator:

```suggestion
  @HostListener('document:click')
  protected _handleClickOutside = e => {
```